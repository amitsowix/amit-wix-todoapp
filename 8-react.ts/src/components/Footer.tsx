import React, {FC} from 'react';
import {createUseStyles} from 'react-jss'
import {Classes} from 'jss';

const Footer: FC = (): JSX.Element => {
    const classes: Classes = useStyles();
    return (
        <div className={classes.appFooter}>
            Created by Amit for Wix.com. All rights reserved Â©
        </div>
    )            
}

const useStyles: () => Classes = createUseStyles({
    appFooter: {
        height: "10%",
        background: "darkred",
        color: "white",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        position: "absolute",
        bottom: "0px",
        width: "100%"
    }
});

export default Footer;