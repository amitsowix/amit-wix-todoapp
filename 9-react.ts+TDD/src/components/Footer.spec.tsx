import React from 'react';
import renderer, {ReactTestRenderer, ReactTestRendererJSON} from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import Footer from './Footer';

const driver = {
    when: {
        mount: (): ReactTestRenderer => {
            const component: ReactTestRenderer = renderer.create(<Footer/>);
            return component;
        },
        render: (): void => {
            render(<Footer/>);
        }
    },
    get: {
      title: () => screen.getByText('Created by Amit for Wix.com. All rights reserved Â©')
    }
}

describe('Footer', () => {
    it('should render properly', () => {

        const component = driver.when.mount();

        let tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should render the correct text', () => {

        driver.when.render();
        
        expect(driver.get.title()).toBeInTheDocument();
    })

    it('should be a div', () => {

        driver.when.render();

        expect(driver.get.title()).toBeInstanceOf(HTMLDivElement);
    })
})
