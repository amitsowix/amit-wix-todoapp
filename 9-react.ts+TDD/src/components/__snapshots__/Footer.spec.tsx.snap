// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render properly 1`] = `
<div
  className="appFooter-0-2-1"
>
  Created by Amit for Wix.com. All rights reserved Â©
</div>
`;
